paddle/memory/detail/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+if(${WITH_GPU})
```

Comment:
It's unnecessary to wrap nv_test in `if(${WITH_GPU})`, because the internal implementation of nv_test already equipped this.